Tests/StoreKitUnitTests/StoreProductTests.swift

```diff
@@ -290,6 +290,11 @@ class StoreProductTests: StoreKitConfigTestCase {
         testSession.locale = Locale(identifier: "en_EN")
         await changeStorefront("USA")
 
+        // Note: this test passes only because the fetcher is recreated
+        // therefore clearing the cache. `ProductsFetcherSK2` does not
+        // detect Storefront changes to invalidate the cache.
```

Comment:
Oh makes sense.
Can you expand this comment to explain that this is done by `StoreKit2StorefrontListenerDelegate`?

Reply:
Also maybe it makes more sense to call `clearCache()` on the fetcher to mimic the real-world behavior?

Reply:
Yep, it's a good point, let me expand it 👍🏼

Reply:
yep, I thought the same, but I wanted to use the same logic for both fetchers. 🤔 BTW, I agree with you that it would mimic the real-world behaviour. Let me change it

Reply:
Done ✅ ([645cf5d](https://github.com/RevenueCat/purchases-ios/pull/1583/commits/645cf5d101a7ecba345dd18749b5734b59a02187)). I've updated the test cases. Now, we clear manually the cache to mimic the real-world behaviour. Moreover, I've updated the comment to indicate that now, the changes of the Storefront are managed by the SK2 listener delegate. 

I think, now it looks better 👍🏼 Thanks!